service/history/workflow/context.go

```diff
@@ -747,12 +747,31 @@ func (c *ContextImpl) mergeUpdateWithNewReplicationTasks(
 	newRunBranchToken := newRunTask.BranchToken
 	newRunID := newRunTask.RunID
 	taskUpdated := false
+
+	updateTask := func(task interface{}) bool {
+		switch t := task.(type) {
+		case *tasks.HistoryReplicationTask:
+			t.NewRunBranchToken = newRunBranchToken
+			t.NewRunID = newRunID
+			return true
+		case *tasks.SyncVersionedTransitionTask:
+			t.NewRunID = newRunID
+			for _, subTask := range t.TaskEquivalents {
+				if historyTask, ok := subTask.(*tasks.HistoryReplicationTask); ok {
+					historyTask.NewRunBranchToken = newRunBranchToken
+					historyTask.NewRunID = newRunID
+					return true
```

Comment:
always return true since task is updated on L758? I mean regardless if there's task equivalents get merged or not?

Reply:
I see. I kinda feel it's coupling the sanity check with the logic but no strong opinion. Maybe add a comment there.